include/sys/sys_heap.h

```diff
@@ -61,6 +61,37 @@ struct z_heap_stress_result {
 	uint64_t accumulated_in_use_bytes;
 };
 
+#ifdef CONFIG_SYS_HEAP_RUNTIME_STATS
+
+struct sys_heap_runtime_stats {
+	size_t free_bytes;
+	size_t allocated_bytes;
+	/* The max available bytes at one time */
+	size_t max_avail_bytes;
+};
+
+/**
+ * @brief Get the runtime statistics of a sys_heap
+ *
+ * @param heap Pointer to specified sys_heap
+ * @param stats_t Pointer to struct to copy statistics into
+ * @return -EINVAL if null pointers, otherwise 0
+ */
+int heap_runtime_stats_get(struct sys_heap *heap,
+		struct sys_heap_runtime_stats *stats);
```

Comment:
Is it a public API? If not, I think we should declare it as a static function.

Reply:
this is a public API, because users need to invoke this function if then wants to know the heap usage.

Reply:
the API sys_heap_runtime_stats_get could be used to get heap usage of any specified sys_heap,
the API heap_runtime_stats_get is only used for the default _system_heap of k_mallock/k_free, 
because for k_malloc, it used the os predefined _system_heap, for malloc, it used the os predifined z_malloc_heap,

@andyross so **do I need to define another two API for them (like heap_runtime_stats_get for k_malloc, using _system_heap by default), or only define a common sys_heap_runtime_stats_get, the users should specific the sys_heap by themselves?**

Reply:
I vote for defining one API - heap_runtime_status_get().

Reply:
I think necessary to try use that API and see if it is convenient and user-friendly.